File: PeoplePicker/AAPLPersonPickerViewController.m
/*
    Copyright (C) 2014 Apple Inc. All Rights Reserved.
    See LICENSE.txt for this sample’s licensing information
    
    Abstract:
    
                This view controller displays the people picker to pick a person.
            
 */

#import "AAPLPersonPickerViewController.h"
@import AddressBook;
@import AddressBookUI;


@interface AAPLPersonPickerViewController () <ABPeoplePickerNavigationControllerDelegate>

@property (weak, nonatomic) IBOutlet UILabel *resultLabel;

@end


@implementation AAPLPersonPickerViewController
            
- (IBAction)showPicker:(id)sender
{
    // This example is to be run on iOS 8.0.
    if ([self isRunningOn8] == NO)
    {
        return;
    }
    
    ABPeoplePickerNavigationController *picker = [[ABPeoplePickerNavigationController alloc] init];
    picker.peoplePickerDelegate = self;
    
    [self presentViewController:picker animated:YES completion:nil];
}


#pragma mark ABPeoplePickerNavigationControllerDelegate methods

// A selected person is returned with this method.
- (void)peoplePickerNavigationController:(ABPeoplePickerNavigationController *)peoplePicker didSelectPerson:(ABRecordRef)person
{
    NSString *contactName = CFBridgingRelease(ABRecordCopyCompositeName(person));
    self.resultLabel.text = [NSString stringWithFormat:@"Picked %@", contactName ? contactName : @"No Name"];
}


// Implement this if you want to do additional work when the picker is cancelled by the user. This method may be optional in a future iOS 8.0 seed.
- (void)peoplePickerNavigationControllerDidCancel:(ABPeoplePickerNavigationController *)peoplePicker
{
}


#pragma mark helper methods

- (BOOL)isRunningOn8
{
    BOOL result = YES;
    NSString *systemVersion = [UIDevice currentDevice].systemVersion;
    if ([systemVersion compare:@"8.0" options:NSNumericSearch] == NSOrderedAscending)
    {
        result = NO;
        UIAlertController *alert = [UIAlertController alertControllerWithTitle:@"Error"
                                                                       message:@"This picker sample can only run\non iOS 8 or later."
                                                                preferredStyle:UIAlertControllerStyleAlert];
        [alert addAction:[UIAlertAction actionWithTitle:@"OK" style:UIAlertActionStyleDefault handler:nil]];
        [self presentViewController:alert animated:YES completion:nil];
    }
    return result;
}

@end

File: PeoplePicker/AAPLPersonPickerViewController.swift
//
//  AAPLPersonPickerViewController.swift
//  PeoplePicker
//
//  Translated by OOPer in cooperation with shlab.jp, on 2014/11/24.
//
//
/*
    Copyright (C) 2014 Apple Inc. All Rights Reserved.
    See LICENSE.txt for this sample’s licensing information

    Abstract:

                The view controller class used for the first tab in our tab bar controller.

 */

import UIKit
import AddressBook
import AddressBookUI


@objc(AAPLPersonPickerViewController)
class PersonPickerViewController: UIViewController, ABPeoplePickerNavigationControllerDelegate {
    
    
    @IBOutlet weak var resultLabel: UILabel!
    
    
    @IBAction func showPicker(_: AnyObject) {
        // This example is to be run on iOS 8.0.
        guard #available(iOS 8.0, *) else {
            showVersionAlert()
            return
        }
        let picker = ABPeoplePickerNavigationController()
        picker.peoplePickerDelegate = self
        
        self.present(picker, animated: true, completion: nil)
    }
    
    
    //#MARK: ABPeoplePickerNavigationControllerDelegate methods
    
    // A selected person is returned with this method.
    func peoplePickerNavigationController(_ peoplePicker: ABPeoplePickerNavigationController, didSelectPerson person: ABRecord) {
        var contactName = ABRecordCopyCompositeName(person)?.takeRetainedValue() as String?
        contactName = contactName ?? "No Name"
        self.resultLabel.text = "Picked \(contactName!)"
    }
    
    
    // Implement this if you want to do additional work when the picker is cancelled by the user. This method may be optional in a future iOS 8.0 seed.
    func peoplePickerNavigationControllerDidCancel(_ peoplePicker: ABPeoplePickerNavigationController) {
    }
    
    
    //#MARK: helper methods
    
    private func showVersionAlert() {
        let alertView = UIAlertView(title: "Error", message: "This picker sample can only run\non iOS 8 or later.", delegate: nil, cancelButtonTitle: "OK")
        alertView.show()
    }
    
}
